### DISCUSSION TITLE
Frequency di Time Series

### DISCUSSION THREAD

**Discussion by takdirzd** [teamalgoritma/community#302] [Q&A] [answered]
permisi kak mau tanya, saya punya data set mulai dari 2014-01-01 sampai 2016-01-01

![image](https://user-images.githubusercontent.com/88231577/189846334-e0c9a1b9-8066-4a34-b392-c3646d83d087.png)

![image](https://user-images.githubusercontent.com/88231577/189846403-5dcd74e4-8de0-436a-aad3-801551bcd3b6.png)

untuk ts() nya saya masukkan seperti ini, saya anggap pola tahunan :
ts(df_new$sale, start = c(2014,1, 1), frequency = 365)
![image](https://user-images.githubusercontent.com/88231577/189848763-eaad6ae3-088e-4867-9979-a691fc31c537.png)

terus saat membuat model say buat seperti ini, saya samakan dengan frequency sebelumnya yaitu 365, tapi tidak bisa di decompose
![image](https://user-images.githubusercontent.com/88231577/189850859-daad04b8-dc16-4e14-8ca7-685210ef35b2.png)

tapi yang terjadi malah error, kalau saya isi 213 baru bisa

======================================================================

trus saat membuat model seperti ini
![image](https://user-images.githubusercontent.com/88231577/189850303-b5eff802-c387-4763-9299-9ee21e86350a.png)

pas forecasting malah jadi garis lurus
![image](https://user-images.githubusercontent.com/88231577/189850457-2d9576ee-a342-43b2-bc7d-3eeb2dce3224.png)

**Comment by anugrah18003** [comment, marked as answer]
Dear @takdirzd 
Penentuan index `start` pada pembuatan objek ts harus berisikan 1 angka atau 2 angka. Contoh penulisannya adalah sebagai berikut:
`ts(data, start = angka1, frequency) `
atau `ts(data, start = c(angka1, angka2) , frequency)`

Sehingga pada kasus diatas ingin mulai dari 2014-01-01 sampai 2016-01-01:
**ts(data,** 
   **start = c(2014,1)**, # dimana angka1 adalah tahun dimulai, angka2 adalah hari ke berapa dalam 365 (setahun disesuaikan dengan frequncy)
   **end = c(2016,1),** # dimana angka1 adalah tahun diakhiri, angka2 adalah hari ke berapa dalam 365 (setahun disesuaikan dengan frequncy)
   **frequency = 365)** # pola seasonal tahunan

======================================================================

Untuk penentuan pada **data train** minimal 2 period, artinya 2x frequency pada kasus diatas maka **2*365**, selain itu hal ini juga menjadi syarat minimal yang diberikan ketika kita akan menggunakan fungsi **decompose()**.

Semoga dapat membantu!

Note: Update 15 September 2022

Referensi: https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/ts

**Reply by takdirzd** [reply]
sudah sy lakukan yg ts()nya, saya pakai end= 2016, 213  , dan berhasil (krn saat saya periksa ts nya memang berakhir di 2016,213 ....

![image](https://user-images.githubusercontent.com/88231577/190844598-ee1d36b1-8603-45b1-8c22-2cb9347f9ebf.png)
tapi untuk kasus forecastnya, saya pakai test 213 , jadi sisanya train pas 2*365 ... tapi pas di visualisasikan malah hasilnya seperti itu, dia mulai forecast di 2017, yang harusnya mulai forecastnya di 2016

**Reply by NabiilahArdini** [reply]
Halo Bapak @takdirzd , sepertinya tampilan hasil forecasting dimulai pada tahun 2017 dikarenakan pemasukan data untuk pemodelan yang salah. Dilakukan forecasting untuk beberapa data setelah data training. Sepertinya Bapak memasukan data training untuk model ets yang mencakup seluruh data (periode train maupun test). 

Selain itu, untuk menjawab pertanyaan mengapa hanya terbentuk garis lurus. Hal ini karena Bapak mengatur parameter "gamma = F" yang membuat model dipaksa tidak menggunakan parameter seasonal yang menangkap pola naik turun berulang dari data. Sedangkan berdasarkan visual, memang terlihat bahwa trend berbentuk lurus. 

Semoga informasi ini dapat membantu.

**Reply by takdirzd** [reply]
baikk terima kasih kak
